FE/src/app/todos/todo-list/todo-list.ts

```diff
@@ -22,23 +25,24 @@ import { MatChipsModule } from '@angular/material/chips';
   templateUrl: './todo-list.html',
   styleUrl: './todo-list.scss',
 })
-export class TodoList {
-  todos = [
-    {
-      id: 1,
-      title: 'Implement new feature',
-      description:
-        'This feature will enhance user experience by providing additional functionality.',
-      type: 'Feature',
-      status: 'Todo',
-      createdOn: new Date('2025-05-26'),
-    },
-    {
-      id: 2,
-      status: 'Done',
-      type: 'Bug',
-      title: 'Fix login issue',
-      createdOn: new Date('2025-05-25'),
-    },
-  ];
+export class TodoList implements OnDestroy {
+  todos$: Observable<Todo[]>;
+  private destroy$ = new Subject<void>();
+  constructor(private todoDataService: TodoDataService) {
+    this.todos$ = this.todoDataService.getTodos();
+  }
+
+  handleDelete(id: number): void {
+    this.todoDataService
+      .deleteTodoReturn(id)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => {
```

Comment:
this will cause a memory leak because it will create a new subscription every time `handleDelete` is called.